core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -203,6 +208,11 @@ class LogManager(val logDirs: Array[File],
                          delay = InitialTaskDelayMs,
                          period = flushCheckpointMs,
                          TimeUnit.MILLISECONDS)
+      scheduler.schedule("kafka-delete-logs",
```

Comment:
I think it is worth observing that an alternate approach to this recurring task that pulls of a queue is to just invoke async delete on demand when we delete a log. i.e., instead of having the `asyncDelete` method and log directory renames, instead call `scheduler.schedule("delete", log.delete)` which will set period -1 to get a single invocation. The reason this isn't very robust is we could get several partitions to delete in a short period of time which could lead to rejected executions if the threadpool is fully utilized.